cudamapper/src/matcher.cu

```diff
@@ -8,107 +8,236 @@
 * license agreement from NVIDIA CORPORATION is strictly prohibited.
 */
 
+#include <algorithm>
+#include <memory>
 #include "matcher.hpp"
+#include <cudautils/cudautils.hpp>
 
 namespace claragenomics {
 
-/*    typedef struct Anchor{
-        read_id_t query_read_id_;
-        read_id_t target_read_id_;
-        position_in_read_t query_position_in_read_;
-        position_in_read_t target_position_in_read_;
-    } Anchor;*/
-
-/*    __global__ void matcher(const position_in_read_t* const positions_in_read_d,
-                            const read_id_t* const read_ids_d,
-                            const ArrayBlock* const blocks_belonging_to_reads_d,
-                            const ArrayBlock* const sketch_elements_with_same_representation_to_check_d,
-                            const ArrayBlock* const sketch_elements_with_same_representation_and_read_d,
-                            Anchor** const anchors_for_read_d,
-                            std:uint32_t* const total_anchors_for_read_d) {
-        extern __shared__ position_in_read_t query_positions_in_read[]; // number of elements == blockDim.x
+    // TODO: move to cudautils?
```

Comment:
Agree with this TODO. Should we move it in this PR?

Reply:
Moved to `cudautils`. Please review new commit and resolve if ok.

Reply:
We kept `unique_ptr_cuda_malloc` in `matcher.cu`. Moving it to `cudautils` will likely happen on the main branch